### PR TITLE
Add feature to remove individual entries from list of recent documents

### PR DESCRIPTION
**Pull request by JayC180**
Added submenu "Remove Recent Documents" under menu file -> recent documents. Clicking on files in this submenu remove individual recent files without the need of clearing the whole list.

### CODE REVIEW

File: src/core/gui/menus/menubar/RecentDocumentsSubmenu.cpp [hunk 1]
```diff
@@ -153,3 +209,11 @@ void RecentDocumentsSubmenu::openFileCallback(GSimpleAction* ga, GVariant* param
...
     self->control->openFile(path);
 }
+
+void RecentDocumentsSubmenu::removeFileCallback(GSimpleAction* ga, GVariant* parameter, RecentDocumentsSubmenu* self) {
+    auto index = g_variant_get_uint64(parameter);
+    auto& path =
+            index < self->xoppFiles.size() ? self->xoppFiles[index] : self->pdfFiles[index - self->xoppFiles.size()];
+    RecentManager::removeRecentFileFilename(path);
+    self->updateMenu();
+}
```
Comment (rolandlo) [inline src/core/gui/menus/menubar/RecentDocumentsSubmenu.cpp:219]: missing newline

File: src/core/gui/menus/menubar/RecentDocumentsSubmenu.cpp [hunk 2]
```diff
@@ -132,6 +169,25 @@ void RecentDocumentsSubmenu::updateMenu() {
     }
     if (this->menuXoppFiles || this->menuPdfFiles) {
         g_menu_append_section(recentFilesSubmenu.get(), nullptr, G_MENU_MODEL(createClearListSection().get()));
+    }
+
+
+    if (!fileList.recentXoppFiles.empty() || !fileList.recentPdfFiles.empty()) {
+        GMenu* removeSubmenu = g_menu_new();
```
Comment (rolandlo) [inline src/core/gui/menus/menubar/RecentDocumentsSubmenu.cpp:176]: Is that data every freed? It doesn't look like it.

File: src/core/gui/menus/menubar/RecentDocumentsSubmenu.cpp [hunk 3]
```diff
@@ -63,6 +80,22 @@ auto createRecentMenu(const container& recentFiles, size_t start_index) -> xoj::
...
+auto createRemoveSubmenu(const container& recentFiles, size_t start_index) -> xoj::util::GObjectSPtr<GMenu> {
+    if (recentFiles.empty()) {
+        return nullptr;
+    }
+    GMenu* menu = g_menu_new();
+
+    for (auto& recent: recentFiles) {
+        auto item = createRemoveMenuItem(recent.get(), start_index);
+        g_menu_append_item(menu, item.get());
+        start_index++;
```
Comment (rolandlo) [inline src/core/gui/menus/menubar/RecentDocumentsSubmenu.cpp:93]: Please keep it consistent with the corresponding line(s) in `createRecentMenu`.

**Review by rolandlo — CHANGES_REQUESTED** [review]
Thanks a lot for your contribution. I have missed that feature as well.

Purely UX-wise (especially when removing multiple recent documents) the Gnome-text-editor way of combining the recent document with a remove icon in a list view would look preferable.
grafik
It would be harder to implement though and reduce consistency among menus.

So I'm fine with the approach you took.

The feature appears to work fine. I have some comments (see in code).
Please also revert the changes to RecentManager.cpp, as they are pure formatting changes unrelated to this PR.

**Comment by JayC180** [comment]
Thank you very much for the review! I have updated the code accordingly.
Regarding the UI suggestion, I also think adding a remove icon would be nice, as that was what I attempted at first. Unfortunately, I am unfamiliar with GTK and not skilled enough to implement that. I recommend creating another issue as enhancement for ui, so it could be picked up by someone with more expertise, and I might attempt that again in the future. Thanks!

**Review by rolandlo — APPROVED** [review]
LGTM. Can you please squash your commits into one?

**Comment by JayC180** [comment]
Done!

**Review by rolandlo — APPROVED** [review]
Thanks for quickly addressing the review.
Merging in 48 hours if no objections are raised.

**Comment by rolandlo** [comment]
It removes individual entries from the list of recent documents. The documents themselves are not removed.

**Comment by JayC180** [comment]
Thanks for the feedbacck! I have added some C_ in RecentDocumentsSubmenu.cpp, hope I did that correctly.

File: src/core/gui/menus/menubar/RecentDocumentsSubmenu.cpp [hunk 4]
```diff
@@ -132,6 +168,28 @@ void RecentDocumentsSubmenu::updateMenu() {
...
+        }
+
+        if (!fileList.recentPdfFiles.empty()) {
+            auto removePdfMenu = createRemoveSubmenu(fileList.recentPdfFiles, fileList.recentXoppFiles.size());
+            g_menu_append_section(removeSubmenu, C_("RemoveInRecentDocumentsSubmenu", "PDF Files"),
+                                  G_MENU_MODEL(removePdfMenu.get()));
+        }
+
+        GMenuItem* removeSubmenuItem = g_menu_item_new_submenu(
+                C_("RemoveInRecentDocumentsSubmenu", "Remove Recent Documents"), G_MENU_MODEL(removeSubmenu));
```
[the remark this answers is not in the document]
Comment (JayC180) [reply src/core/gui/menus/menubar/RecentDocumentsSubmenu.cpp]: Thank you for the explaination! I have updated the file with a more thorough context.

**Pull request merged**